core/src/main/scala/io/projectglow/functions.scala

```diff
@@ -341,6 +341,19 @@ object functions {
     new io.projectglow.sql.expressions.SampleGqSummaryStatistics(genotypes.expr)
   }
 
+  /**
+   * Asserts a boolean condition is true.
+   * @group quality_control
+   * @since 0.5.0
+   *
+   * @param condition Boolean condition to check
+   * @param errMsg Error message if condition fails
+   * @return Null if true, or throws an exception if not true
+   */
+  def assert_true_or_error(condition: Column, errMsg: String): Column = withExpr {
```

Comment:
Hm, actually I think it might be better to exclude this from the Python / Scala APIs. I think the use case is pretty narrow, and since this will be in the next release of Spark, I don't think there's much upside to exposing it in our public API.

You can set the `exclude_python` option in the YAML file. We should add an analogous `exclude_scala` flag. I can take care of that if you don't have time.